services/QuillLMS/lib/tasks/temporary/users.rake

```diff
@@ -20,4 +20,13 @@ namespace :users do
 
     puts "\nUpdated #{num_updates} user emails"
   end
-end
\ No newline at end of file
+
+  task resolve_dual_google_id_and_clever_id: :environment do
+    puts "Going to resolve #{users.count} google_id and clever_id"
+    ActiveRecord::Base.transaction do
```

Comment:
Putting all of these in one transaction (80K rows, several writes each, plus code logic runtime), might lock rows long enough to cause some errors if these users hit the site (it might be fine, I'm not sure how fast this will be). We might consider putting a transaction around smaller sections (e.g. with each `find_in_batches` instead of `find_each`).

It might also be worth asking about whether we need a transaction in general.

Reply:
Actually, given that resolved issues won't be found with the `where` clauses in subsequent runs, this should be idempotent without the transaction block.

Reply:
@dandrabik Good point about transactions and whether they are necessary.  @anathomical good call on this being safe for running again if it fails at some point.  I'll remove the transaction block.